src/jit/morph.cpp

```diff
@@ -19624,3 +19534,136 @@ bool Compiler::fgMorphCombineSIMDFieldAssignments(BasicBlock* block, GenTreePtr
 }
 
 #endif // FEATURE_SIMD
+
+#if !defined(FEATURE_CORECLR) && defined(_TARGET_AMD64_)
+GenTreeStmt* skipNopStmts(GenTreeStmt* stmt)
+{
+    while (stmt != nullptr)
+    {
+        if (!stmt->IsNothingNode())
+        {
+            break;
+        }
+        stmt = stmt->gtNext;
+    }
+}
+
+#endif // !FEATURE_CORECLR && _TARGET_AMD64_
+
+bool Compiler::fgCheckStmtAfterTailCall()
```

Comment:
Please add a header comment.

It would be helpful if the comment summarized the "expected" types of instructions that can follow a tail call in a block.

Reply:
Fixed, but the patterns are described in the body.